cg/meta/upload/mutacc.py

```diff
@@ -0,0 +1,194 @@
+import logging
+from collections import namedtuple
+
+from cg.apps import scoutapi, mutacc_auto
+
+LOG = logging.getLogger(__name__)
+
+
+class UploadToMutaccAPI():
+
+    """API to upload finished cases to mutacc"""
+
+    def __init__(self, scout_api: scoutapi.ScoutAPI, mutacc_auto_api: mutacc_auto.MutaccAutoAPI):
+
+        self.scout = scout_api
+        self.mutacc_auto = mutacc_auto_api
+
+    def _data(self, case) -> dict:
+        """
+            Find the necessary data for the case
+
+            Args:
+                case (dict): case dictionary from scout
+
+            Returns:
+                data (dict): dictionary with case data, and data on causative variants
+        """
+
+        if all([self._has_bam(case), self._has_causatives(case)]):
+            causatives = self.scout.get_causative_variants(case_id=case['_id'])
+            mutacc_case = remap(case, SCOUT_TO_MUTACC_CASE)
+            mutacc_variants = [remap(variant, SCOUT_TO_MUTACC_VARIANTS) for variant in causatives]
+            return {'case': mutacc_case, 'causatives': mutacc_variants}
+        return {}
+
+    def extract_reads(self, case: dict):
+        """Use mutacc API to extract reads from case"""
+        data = self._data(case)
+        if data:
+            LOG.info("Extracting reads from case %s", case['_id'])
+            self.mutacc_auto.extract_reads(case=data['case'], variants=data['causatives'])
+
+    def import_cases(self):
+        """Use mutacc API to import cases to database"""
+        LOG.info('importing cases into mutacc database')
+        self.mutacc_auto.import_reads()
+
+    @staticmethod
+    def _has_bam(case: dict) -> bool:
+
+        """
+            Check that all samples in case has a given path to a bam file,
+            and that the file exists
+
+            Args:
+                case (dict): case dictionary from scout
+
+            Returns:
+                (bool): True if all samples has valid paths to a bam-file
+
+        """
+
+
+        for sample in case['individuals']:
+
+            if sample.get('bam_file', None) is None:
+                LOG.warning("sample %s in case %s is missing bam fille",
+                            sample['individual_id'], case['_id'])
+                return False
+
+            elif not os.path.isfile(sample['bam_file']):
+                LOG.warning("sample %s in %s has non existing bam_file",
+                            sample['individual_id'], case['_id'])
+                return False
+
+        return True
+
+    @staticmethod
+    def _has_causatives(case: dict) -> bool:
+        """
+            Check that the case has marked causative variants in scout
+
+            Args:
+                case (dict): case dictionary from scout
+
+            Returns:
+                (bool): True if case has marked causative variants in scout
+        """
+        if case.get('causatives'):
+            return True
+
+        LOG.warning("case %s has no marked causatives in scout", case['_id'])
```

Comment:
why is this a warning? don't we ever expect to get here?

Reply:
will change log level to